pkg/operator/controller/gatewayclass/subscription_test.go

```diff
@@ -0,0 +1,416 @@
+package gatewayclass
+
+import (
+	"context"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorcontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func Test_ensureServiceMeshOperatorInstallPlan(t *testing.T) {
+	tests := []struct {
+		name            string
+		channel         string
+		version         string
+		existingObjects []runtime.Object
+		expectCreate    []client.Object
+		expectUpdate    []client.Object
+		expectDelete    []client.Object
```

Comment:
We would never delete an install plan, right?  So we don't need `expectDelete`.

Reply:
We never create or delete install plans, but verifying that nothing is created or deleted is still a helpful test signal. If there are other changes that are needed, I can update the test to get rid of the `expectCreate` and `expectDelete` variables, and always check that there are no creates or deletes, but I think this accomplishes the same thing in a similarly readable way.